### ISSUE TITLE
[BUG] 'file.tidied' does not honor the 'time_comparison' and 'age' parameters

### ISSUE THREAD

**Issue by chrisvmw** [saltstack/salt#63019]
**Description**
I am trying to use the `file.tidied` state to remove files that have a modification time greater than 30 days.  Instead, it is removing all files in the target directory, regardless of age.

**Setup**

This is a fresh "classic" manual install of both Salt and the minion, running version 3005.1.

State file:

```
[root@ip-10-0-4-208 project03]# cat tidy_test.sls
cleanup:
  file.tidied:
    - name: /var/crash-test
    - age: 30
    - time_comparison: mtime
    - rmdirs: True
    - exclude:
      - lost*
[root@ip-10-0-4-208 project03]#
```

State run, and output showing that files that have modification times less than and greater than 30 days will be removed.

```
[root@ip-10-0-4-208 project03]# salt -vvv rhel8-3 state.apply project03.tidy_test test=True
Executing job with jid 20221104193605436535
-------------------------------------------

rhel8-3:
----------
          ID: cleanup
    Function: file.tidied
        Name: /var/crash-test
      Result: None
     Comment: /var/crash-test is set for tidy
     Started: 19:36:05.790885
    Duration: 1.773 ms
     Changes:
              ----------
              removed:
                  - /var/crash-test/foo
                  - /var/crash-test/hosts
                  - /var/crash-test/largefile
                  - /var/crash-test/resolv.conf
                  - /var/crash-test/mydir

Summary for rhel8-3
------------
Succeeded: 1 (unchanged=1, changed=1)
Failed:    0
------------
Total states run:     1
Total run time:   1.773 ms
[root@ip-10-0-4-208 project03]# vim tidy_test.sls
[root@ip-10-0-4-208 project03]# salt rhel8-3 cmd.run 'ls -l /var/crash-test'
rhel8-3:
    total 9539652
    -rw-r--r--. 1 root root        158 Oct 19  2012 foo
    -rw-r--r--. 1 root root        158 Jan 18  2019 hosts
    -rw-r--r--. 1 root root 9768534016 Nov  3 21:48 largefile
    drwx------. 2 root root      49152 Nov  2 21:17 lost+found
    drwxr-xr-x. 2 root root       4096 Nov  3 21:19 mydir
    -rw-r--r--. 1 root root         84 Nov  3 16:37 resolv.conf
[root@ip-10-0-4-208 project03]#
[root@ip-10-0-4-208 project03]# salt rhel8-3 cmd.run 'stat /var/crash-test/hosts'
rhel8-3:
      File: /var/crash-test/hosts
      Size: 158       	Blocks: 8          IO Block: 4096   regular file
    Device: ca11h/51729d	Inode: 11          Links: 1
    Access: (0644/-rw-r--r--)  Uid: (    0/    root)   Gid: (    0/    root)
    Context: unconfined_u:object_r:unlabeled_t:s0
    Access: 2019-01-18 12:05:09.000000000 +0000
    Modify: 2019-01-18 12:05:09.000000000 +0000
    Change: 2022-11-03 16:51:39.460954098 +0000
     Birth: 2022-11-03 16:47:37.159151752 +0000
[root@ip-10-0-4-208 project03]#
```

Salt master:

```
[root@ip-10-0-4-208 project03]#  salt-run --versions-report
Salt Version:
          Salt: 3005.1

Dependency Versions:
          cffi: Not Installed
      cherrypy: Not Installed
      dateutil: Not Installed
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 2.11.1
       libgit2: Not Installed
      M2Crypto: 0.35.2
          Mako: Not Installed
       msgpack: 0.6.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: Not Installed
      pycrypto: Not Installed
  pycryptodome: Not Installed
        pygit2: Not Installed
        Python: 3.6.8 (default, Nov 16 2020, 16:55:22)
  python-gnupg: Not Installed
        PyYAML: 3.13
         PyZMQ: 18.0.1
         smmap: Not Installed
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.1.4

System Versions:
          dist: centos 7 Core
        locale: UTF-8
       machine: x86_64
       release: 3.10.0-957.1.3.el7.x86_64
        system: Linux
       version: CentOS Linux 7 Core

[root@ip-10-0-4-208 project03]#
```

Minion:

```
[root@ip-10-0-211-211 ~]# salt-call --local test.versions
local:
    Salt Version:
              Salt: 3005.1

    Dependency Versions:
              cffi: 1.14.6
          cherrypy: 18.6.1
          dateutil: 2.8.1
         docker-py: Not Installed
             gitdb: Not Installed
         gitpython: Not Installed
            Jinja2: 3.1.0
           libgit2: Not Installed
          M2Crypto: Not Installed
              Mako: Not Installed
           msgpack: 1.0.2
      msgpack-pure: Not Installed
      mysql-python: Not Installed
         pycparser: 2.21
          pycrypto: Not Installed
      pycryptodome: 3.9.8
            pygit2: Not Installed
            Python: 3.9.14 (main, Oct  3 2022, 21:24:00)
      python-gnupg: 0.4.8
            PyYAML: 5.4.1
             PyZMQ: 23.2.0
             smmap: Not Installed
           timelib: 0.2.4
           Tornado: 4.5.3
               ZMQ: 4.3.4

    System Versions:
              dist: rhel 8.6 Ootpa
            locale: utf-8
           machine: x86_64
           release: 4.18.0-372.9.1.el8.x86_64
            system: Linux
           version: Red Hat Enterprise Linux 8.6 Ootpa

[root@ip-10-0-211-211 ~]#
```

Please be as specific as possible and give set-up details.

- [X ] VM running on a cloud service, please be explicit and add details
- [ X] classic packaging

Both Salt master and minion are running in AWS.

**Steps to Reproduce the behavior**
(Include debug logs if possible and relevant)

See above.

**Expected behavior**

With the parameters `age: 30` and `time_comparison: mtime`,  the newer files (such as 'largefile' and 'resolv.conf' should not be removed.

**Comment by OrangeDog** [comment]
Duplicate: #62955

**Comment by dmurphy18** [comment]
@chrisvmw Closing this as duplicate and fix will be in the next major release, 3006.0
